server/context.go

```diff
@@ -10,16 +10,16 @@ import (
 )
 
 // SessionBuilder creates sessions given a MySQL connection and a server address.
-type SessionBuilder func(conn *mysql.Conn, addr string) sql.Session
+type SessionBuilder func(conn *mysql.Conn, addr string) (sql.Session, *sql.IndexRegistry, *sql.ViewRegistry)
```

Comment:
When you build a session you also provide the IndexRegistry and ViewRegistry that should be used with that session.